### ISSUE TITLE
LigerCrossEntropyLoss is not patched for latest transformers models

### ISSUE THREAD

**Issue by Tcc0403** [linkedin/Liger-Kernel#369]
### 🐛 Describe the bug

To fix GA bug, latest transformers no longer imports CrossEntropyLoss. Instead, it's wrapped in [`self.loss_function`](https://github.com/huggingface/transformers/blob/a06a0d12636756352494b99b5b264ac9955bc735/src/transformers/models/llama/modeling_llama.py#L1224) which can be traced back to [here](https://github.com/huggingface/transformers/blob/a06a0d12636756352494b99b5b264ac9955bc735/src/transformers/loss/loss_utils.py#L26). So that [current patching method](https://github.com/linkedin/Liger-Kernel/blob/2d3eb94679cec946c8f85a15590cbef9c1327250/src/liger_kernel/transformers/monkey_patch.py#L115) doesn't work on latest transformers model.

Note that [labels and logits shifting operations are wrapped](https://github.com/huggingface/transformers/blob/a06a0d12636756352494b99b5b264ac9955bc735/src/transformers/loss/loss_utils.py#L32) in `loss_function` as well, we need to take care of it when implementing new patching method.

### Versions

none

**Comment by ByronHsu** [comment]
fixed by https://github.com/linkedin/Liger-Kernel/pull/375
